ldap/servers/slapd/back-ldbm/ldbm_search.c

```diff
@@ -1515,6 +1517,18 @@ ldbm_back_next_search_entry(Slapi_PBlock *pb)
 
     /* Find the next candidate entry and return it. */
     while (1) {
+        if (li->li_dblock_monitoring &&
+            slapi_atomic_load_32(&(li->li_dblock_threshold_reached), __ATOMIC_RELAXED)) {
+            errmsg = PR_smprintf("DB locks threshold is reached (nsslapd-db-locks-monitoring-threshold "
+                                 "under cn=bdb,cn=config,cn=ldbm database,cn=plugins,cn=config). "
+                                 "Please, increase nsslapd-db-locks according to your needs.");
+            slapi_log_err(SLAPI_LOG_CRIT, "ldbm_back_next_search_entry", "%s\n", errmsg);
+            slapi_pblock_set(pb, SLAPI_SEARCH_RESULT_ENTRY, NULL);
+            delete_search_result_set(pb, &sr);
+            rc = SLAPI_FAIL_GENERAL;
+            slapi_send_ldap_result(pb, LDAP_UNWILLING_TO_PERFORM, NULL, errmsg, 0, NULL);
```

Comment:
I would not send this detailed error message to the ldapclient. rather something like "Transient DB ressource critical level".

Reply:
Good idea, I'll add it with my next commit.